quic/s2n-quic-events/src/parser.rs

```diff
@@ -282,6 +283,25 @@ impl Struct {
                         }
                     ));
 
+                    // Metrics is only connection-level events
+                    output.metrics_fields.extend(quote!(
+                        pub #counter: #counter_type,
+                    ));
+                    output.metrics_fields_init.extend(quote!(
+                        #counter: #counter_init,
+                    ));
+
+                    output.metrics_record.extend(quote!(
+                        self.recorder.increment_counter(#snake, self.#counter #counter_load as _);
+                    ));
+
+                    output.subscriber_metrics.extend(quote!(
+                        #allow_deprecated
```

Comment:
We should probably inline, since the impl is super cheap

```suggestion
                        #[inline]
                        #allow_deprecated
```